### DISCUSSION TITLE
sealed 클래스에 대해 설명해 주세요.

### DISCUSSION THREAD

**Discussion by 3Juhwan** [2024-woowacourse-study/level-interview#90] [Java]
.

**Comment by gyungchan-jo** [comment]
Java 17부터 도입된 Sealed 클래스는 한정된 하위 클래스를 갖도록 클래스를 정의하는 방법입니다. Sealed 클래스를 정의하면 해당 클래스의 하위 클래스를 명시적으로 지정할 수 있습니다. 이는 상속을 제한하고 클래스의 불변성과 안정성을 유지하는 데 도움이 됩니다.

Sealed 클래스를 정의하려면 클래스 선언 앞에 sealed 키워드를 사용하고, 허용되는 하위 클래스를 permits 키워드를 사용하여 지정합니다. Sealed 클래스의 모든 하위 클래스는 명시적으로 extends 키워드를 사용하여 선언되어야 합니다.

**Comment by le2sky** [comment]
sealed 키워드는 Java 17에서 추가되었습니다. (인터페이스나 클래스를) 구현하거나 상속받을 클래스를 미리 지정해두고, 그 이외 클래스는 상속 및 구현을 불가능하도록 제한할 수 있습니다. 이를 통해, 해당 키워드가 붙은 클래스나 인터페이스를 누가 상속받는지 명확히 인지할 수 있으며, 제한 가능합니다. 

(sealed 클래스 기준으로 설명)
super 클래스에는 sealed 키워드를 사용하며, permits 키워드 뒤에 해당 클래스를 상속받을 sub 클래스를 선언합니다. 

```
public sealed class Crew permits Atom, Pororo {}
```

sealed 키워드의 하위 클래스는 명시적으로 sealed, non-sealed, final 키워드를 지정해줘야합니다.
non-sealed 키워드는 어느 클래스나 상속받을 수 있음을 의미합니다.
final은 어떤 클래스도 상속 불가능함을 의미합니다.
